pytorch_lightning/trainer/configuration_validator.py

```diff
@@ -24,18 +24,16 @@ def __init__(self, trainer):
 
     def verify_loop_configurations(self, model: LightningModule):
         r"""
-        Checks that the model is configured correctly before training or testing is started.
+        Checks that the model is configured correctly before the run is started.
 
         Args:
             model: The model to check the configuration.
 
         """
-        if not self.trainer.testing:
+        if self.trainer.training:
```

Comment:
Someone probably asked this already, sorry but I have to ask again. 
The fact that you replace `not trainer.testing` with `trainer.training` here but in other places you replace it with `TrainerState.TRAINING` is bothering me and I just can't wrap my head around this. 

Looking at all code changes, and the new definitions, it seems like it roughly works like this:

a) TrainerState strictly tells us if the user called fit(), test(), tune(). 
This state then doesn't change until the routine is over. 
b) There is the RunningStage, which can change within a fit() call from training to validating and so on.

Are these observations correct?

Reply:
Absolutely correct!

For example, validation can be run both during `trainer.fit` (`TrainerState.FITTING`) and `trainer.validate` (`TrainerState.VALIDATING`). Having both we can properly separate this logic.

For tuning, predicting, testing. It does not matter currently as we don't run anything else than the `RunningStage` with the same name